pkg/ruler/remotequerier.go

```diff
@@ -253,6 +254,27 @@ func (q *RemoteQuerier) query(ctx context.Context, query string, ts time.Time, l
 	return v.Type, v.Result, nil
 }
 
+func (q *RemoteQuerier) sendRequest(ctx context.Context, req *httpgrpc.HTTPRequest) (*httpgrpc.HTTPResponse, error) {
+	// Ongoing request may be cancelled during evaluation due to some transient error or server shutdown,
+	// so we'll keep retrying until we get a successful response or context is eventually cancelled.
+	retryConfig := backoff.Config{
```

Comment:
I think it's not very safe to just rely on context cancellation. I would set a max retries too. You may consider to set something like max 3 retries, but increase the min backoff to few seconds.

Reply:
Agreed. The default timeout that seems to be used for this context is 2 minutes which is a lot of retires.

Reply:
Yeah, this makes sense. 👍 

Changed @ 94341fbe0ec1d1d622c23b7be607c784a0bb68ad